src/CMakeLists.txt

```diff
@@ -37,17 +37,29 @@ initialize_lto_property()
 # The module will be added to the project as a library built from the source
 # files (*.cpp) available in the current directory.
 #
+# All configurable header files (*.hpp.in) and files with configurable template
+# implementations (*.tpp.in) will be automatically configured and copied into
+# the build directory.
+#
 # All header files (*.hpp) and files with template implementations (*.tpp) in
 # the current directory and in the build directory will be installed inside
-# the directory "include/${PROJECT_NAME}". In order for the files in the build
-# directory to be properly installed they need to be created before calling
-# this function.
+# the directory "include/${PROJECT_NAME}". Only files automatically generated
+# by this function (e.g., configurable header files) or files created before
+# calling this function will be installed.
 function(configureModule MODULE_NAME)
     string(TOUPPER ${MODULE_NAME} UPPER_MODULE_NAME)
 
     # Configure compilation
     addModuleIncludeDirectories(${MODULE_NAME})
 
+    # Configure headers
+    file(GLOB CONFIGURABLE_HEADER_FILES "*.hpp.in" "*.tpp.in")
+    foreach(CONFIGURABLE_HEADER_FILE IN LISTS CONFIGURABLE_HEADER_FILES)
+        string(REGEX REPLACE "\\.[^.]*$" "" CONFIGURED_HEADER_FILE ${CONFIGURABLE_HEADER_FILE})
+        set(CONFIGURED_HEADER_FILE "${PROJECT_BINARY_DIR}/src/${MODULE_NAME}/${CONFIGURED_HEADER_FILE}")
```

Comment:
I think that is the intent. My original proposal (still on the table) was to configure a header (hpp.in) to generate a configured header (.hpp). Then copy the configured header in the build directory (in the corresponding module's subdir). Then at installation time, all bitpit's header would be copied in the (user's defined) installation dir as usual. I think step 2 has been removed (but I can't remember why).

Reply:
> There is a problem getting the name of the final configured file, that has to be moved in ${PROJECT_BINARY_DIR}/src/${MODULE_NAME} directory.
> file(glob )return a list of absolute path to *in template files, string(REGEX REPLACE ) for each entry purge the .in extension only, but the whole directory path is mantained in the name.
> The final result in CONFIGURED_HEADER_FILE is *_${PROJECT_BINARY_DIR}/src/${MODULE_NAME}/< full dir path of glob name > / < purged name without _in >__ instead of the expected result *_${PROJECT_BINARY_DIR}/src/${MODULE_NAME}/ < purged name without _in >__

Fixed. The original implementation was using cmake_path (introduced in cmake 3.20), I introduced the problem when I tried to replace that function with get_filename_component (available since cmake 2.8).

Reply:
> I think that is the intent. My original proposal (still on the table) was to configure a header (hpp.in) to generate a configured header (.hpp). Then copy the configured header in the build directory (in the corresponding module's subdir). Then at installation time, all bitpit's header would be copied in the (user's defined) installation dir as usual. I think step 2 has been removed (but I can't remember why).

Configurable headers (hpp.in and tpp.in) are configured and then copied in the build directory. During installation, all headers inside build and source directories are copied in the user's defined installation path. What this patch series is not doing is copy non-configurable headers in the build directory (they will be copied in the user's defined installation path directly from the source directory).

Reply:
> > There is a problem getting the name of the final configured file, that has to be moved in ${PROJECT_BINARY_DIR}/src/${MODULE_NAME} directory.
> > file(glob )return a list of absolute path to *in template files, string(REGEX REPLACE ) for each entry purge the .in extension only, but the whole directory path is mantained in the name.
> > The final result in CONFIGURED_HEADER_FILE is *_${PROJECT_BINARY_DIR}/src/${MODULE_NAME}/< full dir path of glob name > / < purged name without _in >__ instead of the expected result *_${PROJECT_BINARY_DIR}/src/${MODULE_NAME}/ < purged name without _in >__
> 
> Fixed. The original implementation was using cmake_path (introduced in cmake 3.20), I introduced the problem when I tried to replace that function with get_filename_component (available since cmake 2.8).

Ok, fixing is fine for me.